portal-bridge/src/constants.rs

```diff
@@ -18,9 +18,16 @@ pub const BEACON_GENESIS_TIME: u64 = 1606824023;
 // overloaded and not performing well
 pub const HTTP_REQUEST_TIMEOUT: Duration = Duration::from_secs(20);
 
-// This is the maximum number of active blocks being gossiped
+// The maximum number of active blocks being gossiped. Note that this doesn't
+// exactly mean the number of concurrent gossip jsonrpc requests, as the gossip
+// requests are batched / block.
+// Currently, this is only used in the history & era1 bridges
 pub const DEFAULT_GOSSIP_LIMIT: usize = 32;
 
+// The maximum number of concurrent offer jsonrpc requests that can be made simultaneously.
+// Currently, this is only used in the state bridge.
+pub const DEFAULT_OFFER_LIMIT: usize = 64;
```

Comment:
Shouldn't this be `32 * 4` or `32 * 8`, as for 1 gossip limit, trin can spawn 8 offers

Reply:
I don't think so, since we're using the the semaphore to limit the spawning of individual offer tasks, not entire blocks. Which, imo is the key distinction between state & (history / era1) bridges.

https://github.com/ethereum/trin/blob/master/portal-bridge/src/bridge/state.rs#L248

Reply:
Or maybe I'm misunderstanding...

Reply:
I think it's cleaner to just decouple the concept of an `offer_limit` from a `gossip_limit` entirely

Reply:
> I think it's cleaner to just decouple the concept of an `offer_limit` from a `gossip_limit` entirely

I think this is the right direction, I just think 64 is way to low

Reply:
Ahh, gotcha... well so far in my testing it seems to hold up pretty well up to a gossip-limit of ~100, but not so much over 150. Haven't exactly nailed down the exact number yet, but I'd say we can just leave it for now and I'll update this value later once I've got a better idea of where the actual limit is

Reply:
Or, I could do 100 if you feel like it's worthwhile, but if a user wants they can always manually increase this limit using a cli flag

Reply:
Ok, I understand your reasoning now, but I suspect that earlier when running a gossip limit of 32 (aka 32 * 4 concurrent rpc requests) that they weren't actually all successful txs

Reply:
> Or, I could do 100 if you feel like it's worthwhile, but if a user wants they can always manually increase this limit using a cli flag

oh ok 64 or 100 is good then up to you